source/ui/PGM/PGMUndulatorView.cpp

```diff
@@ -0,0 +1,18 @@
+#include "PGMUndulatorView.h"
+
+#include <QVBoxLayout>
+
+#include "ui/beamline/AMExtendedControlEditor.h"
+#include "beamline/PGM/PGMBeamline.h"
+
+PGMUndulatorView::PGMUndulatorView(QWidget *parent) : QWidget(parent)
```

Comment:
at this moment, this class is very alike `PGMGratingView`. Maybe we should think to generalize them?